src/DataTable/utils/getVisibleColumns.jsx

```diff
@@ -21,7 +23,7 @@ export const selectColumn = {
     );
     const updatedProps = useConvertIndeterminateProp(toggleRowsSelectedProps);
 
-    return (
+    return isSelectable && maxSelectedRows ? null : (
```

Comment:
Would it be possible to split this logic up into explicit `if` statements? Might be a bit more readable. E.g.,

```jsx
if (isSelectable && maxSelectedRows) {
  return null;
}

return (
  ...
);
```